### PR TITLE
Add crossfade animation during orientation change when used within UIKit hierarchy

### PR DESCRIPTION
**Pull request by elijah-semyonov**
## Proposed Changes

Add crossfade animation during orientation change to avoid stretching (Compose doesn't support per-view frame interpolation like native UIKit does)

## Testing

Test: instantiate ComposeWindow outside of SwiftUI UIViewControllerRepresentable context, rotate the device.

### Before

https://github.com/JetBrains/compose-multiplatform-core/assets/4167681/4cfaa3c4-f448-4e8c-8502-127ac24641e0

### After

https://github.com/JetBrains/compose-multiplatform-core/assets/4167681/2304d8b0-f9c1-4021-86a6-830ec50c8859

https://github.com/JetBrains/compose-multiplatform-core/assets/4167681/305866ca-5652-417b-bfc6-85d30f2f2801

### CODE REVIEW

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt [hunk 1]
```diff
@@ -94,6 +99,41 @@ private class AttachedComposeContext(
...
+
+    private fun setConstraints(newConstraints: List<NSLayoutConstraint>) {
+        if (constraints.isNotEmpty()) {
+            NSLayoutConstraint.deactivateConstraints(constraints)
+        }
+
+        constraints = newConstraints
+        NSLayoutConstraint.activateConstraints(newConstraints)
+    }
+
```
Comment (dima-avdeev-jb) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt]:
```suggestion
    private var constraints: List<NSLayoutConstraint> = emptyList()
        set(value) {
            if (field.isNotEmpty()) {
                NSLayoutConstraint.deactivateConstraints(field)
            }
            field = value
            NSLayoutConstraint.activateConstraints(value)
        }

    fun setConstraintsToCenterInView(parentView: UIView, size: CValue<CGSize>) {
        size.useContents {
            constraints = listOf(
                view.centerXAnchor.constraintEqualToAnchor(parentView.centerXAnchor),
                view.centerYAnchor.constraintEqualToAnchor(parentView.centerYAnchor),
                view.widthAnchor.constraintEqualToConstant(width),
                view.heightAnchor.constraintEqualToConstant(height)
            )
        }
    }

    fun setConstraintsToFillView(parentView: UIView) {
        constraints = listOf(
            view.leftAnchor.constraintEqualToAnchor(parentView.leftAnchor),
            view.rightAnchor.constraintEqualToAnchor(parentView.rightAnchor),
            view.topAnchor.constraintEqualToAnchor(parentView.topAnchor),
            view.bottomAnchor.constraintEqualToAnchor(parentView.bottomAnchor)
        )
    }
```
setConstraints function can be simplifyed to variable property with custom setter (doc https://kotlinlang.org/docs/properties.html#getters-and-setters)

**Comment by dima-avdeev-jb** [comment]
Found janks in our Demo project on simulator:


https://github.com/JetBrains/compose-multiplatform-core/assets/99798741/748d378f-61b0-4dfe-8ece-8b5dc4fdb8be

**Comment by elijah-semyonov** [comment]
> Found janks in our Demo project on simulator:

Fixed in https://github.com/JetBrains/compose-multiplatform-core/pull/778/commits/0a22390fef282cd75cdc429739e37f98cec3ae00

**Review by dima-avdeev-jb — APPROVED** [review]

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt [hunk 2]
```diff
@@ -560,11 +654,33 @@ internal actual class ComposeWindow : UIViewController {
...
         attachedComposeContext =
             AttachedComposeContext(scene, skikoUIView).also {
+                it.setConstraintsToFillView(view)
                 updateLayout(it)
             }
     }
 }
 
+private fun UIViewController.checkIfInsideSwiftUI(): Boolean {
+        var parent = parentViewController
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt:664]: This part is not formatter properly

**Pull request merged**